nucypher/network/server.py

```diff
@@ -104,29 +103,25 @@ def __init__(self, *args, **kwargs):
 class ProxyRESTServer:
 
     def __init__(self,
-                 host=None,
-                 port=None,
+                 rest_host=None,
+                 rest_port=None,
                  db_name=None,
                  tls_private_key=None,
                  tls_curve=None,
-                 *args, **kwargs):
-        self.rest_interface = InterfaceInfo(host=host, port=port)
-
+                 certificate=None,
+                 *args, ** kwargs) -> None:
+        self.rest_interface = InterfaceInfo(host=rest_host, port=rest_port)
         self.db_name = db_name
         self._rest_app = None
         tls_hosting_keypair = HostingKeypair(common_name=self.checksum_public_address,
-                                             private_key=tls_private_key, curve=tls_curve)
+                                             private_key=tls_private_key,
+                                             curve=tls_curve,
+                                             host=rest_host,
+                                             certificate=certificate)
         tls_hosting_power = TLSHostingPower(keypair=tls_hosting_keypair)
         self._crypto_power.consume_power_up(tls_hosting_power)
 
-    @classmethod
-    def from_config(cls, network_config: NetworkConfiguration = None):
-        """Create a server object from config values, or from a config file."""
-        # if network_config is None:
-        # NetworkConfiguration._load()
-        instance = cls()
-
-    def public_key(self, power_class: ClassVar):
+    def public_material(self, power_class: ClassVar):
         """Implemented on Ursula"""
         raise NotImplementedError
```

Comment:
This is very weird. Perhaps removable?

Reply:
I'm :+1: on removing it, but it is widely regarded as an OOP antipattern to do so.